pvlib/temperature.py

```diff
@@ -423,3 +424,199 @@ def faiman(poa_global, temp_air, wind_speed=1.0, u0=25.0, u1=6.84):
     heat_input = poa_global
     temp_difference = heat_input / total_loss_factor
     return temp_air + temp_difference
+
+
+def _fuentes_hconv(tave, windmod, tinoct, temp_delta, xlen, check_reynold):
+    # Calculate the convective coefficient as in Fuentes 1987 -- a mixture of
+    # free, laminar, and turbulent convection.
+    densair = 0.003484 * 101325.0 / tave  # density
+    visair = 0.24237e-6 * tave**0.76 / densair  # kinematic viscosity
+    condair = 2.1695e-4 * tave**0.84  # thermal conductivity
+    reynold = windmod * xlen / visair
+    # the boundary between laminar and turbulent is modeled as an abrupt
+    # change at Re = 1.2e5:
+    if check_reynold and reynold > 1.2e5:
+        # turbulent convection
+        hforce = 0.0282 / reynold**0.2 * densair * windmod * 1007 / 0.71**0.4
+    else:
+        # laminar convection
+        hforce = 0.8600 / reynold**0.5 * densair * windmod * 1007 / 0.71**0.67
+    # free convection via Grashof number
+    # NB: the 0.5 factor is from assuming tilt=30; should tilt be a parameter?
+    grashof = 9.8 / tave * temp_delta * xlen**3 / visair**2 * 0.5
+    # product of Nusselt number and (k/l)
+    hfree = 0.21 * (grashof * 0.71)**0.32 * condair / xlen
+    # combine free and forced components
+    hconv = (hfree**3 + hforce**3)**(1/3)
+    return hconv
+
+
+def fuentes(poa_global, temp_air, wind_speed, noct_installed, module_height=5,
+            wind_height=9.144, emissivity=0.84, absorption=0.83,
+            hydraulic_diameter=0.5):
+    """
+    Calculate cell or module temperature using the Fuentes model.
+
+    The Fuentes model is a first-principles heat transfer energy balance
+    model [1]_ that is used in PVWatts for cell temperature modeling [2]_.
+
+    Parameters
+    ----------
+    poa_global : pandas Series
+        Total incident irradiance [W/m^2]
+
+    temp_air : pandas Series
+        Ambient dry bulb temperature [C]
+
+    wind_speed : pandas Series
+        Wind speed [m/s]
+
+    noct_installed : float
+        The "installed" nominal operating cell temperature as defined in [1]_.
+        PVWatts assumes this value to be 45 C for rack-mounted arrays and
+        49 C for roof mount systems with restricted air flow around the
+        module.  [C]
+
+    module_height : float, default 5.0
+        The height above ground of the module above the ground. The PVWatts
+        default is 5.0 [m]
+
+    wind_height : float, default 9.144
+        The height above ground at which ``wind_speed`` is measured. The
+        PVWatts defauls is 9.144 [m]
+
+    emissivity : float, default 0.84
+        The effectiveness of the module at radiating thermal energy. [unitless]
+
+    absorption : float, default 0.83
+        The fraction of incident irradiance that is converted to thermal
+        energy in the module. [unitless]
+
+    hydraulic_diameter : float, default 0.5
```

Comment:
Should have a modern default for a 60 or 72 cell, 6-inch cell module. Fuentes leaves out a lot of detail. Maybe replace `hydraulic_diameter` with `module_width`, `module_length` and calculate the diameter?

Reply:
That would make the API nicer, but using modern defaults would make the default arguments inconsistent with PVWatts, which as far as I can tell follows the Fuentes defaults.  Which should take precedence here?

Reply:
I guess, keep the default to mimic PVWatt's behavior now, and add to the docstring how to calculate hydraulic diameter for a rectangular module.

Reply:
Oh, I missed the last comment here -- I changed it to take `module_width` and `module_length` as parameters and calculate the hydraulic diameter internally.  Happy to change it back and add a Notes section with the hydraulic diameter equation if you think that's the better option, let me know.

Reply:
No need to change it back, it's was OK before and better now.